From stable-bounces@linux.kernel.org Tue Oct  3 00:16:13 2006
From: Neil Brown <neilb@suse.de>
To: <syrius.ml@no-log.org>, "Richard Bollinger" <rabollinger@gmail.com>
Date: Tue, 3 Oct 2006 17:15:00 +1000
Message-ID: <17698.3572.190300.504962@cse.unsw.edu.au>
Cc: linux-raid@vger.kernel.org, stable@kernel.org
Subject: MD: Fix problem where hot-added drives are not resynced.

From: Neil Brown <neilb@suse.de>

If a drive is added with HOT_ADD_DISK rather than ADD_NEW_DISK,
saved_raid_disk isn't initialised properly, and the drive can be
included in the array without a resync.


From: Neil Brown <neilb@suse.de>
Cc: <syrius.ml@no-log.org>
Cc: Richard Bollinger <rabollinger@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/md/md.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.18.orig/drivers/md/md.c
+++ linux-2.6.18/drivers/md/md.c
@@ -3867,6 +3867,7 @@ static int hot_add_disk(mddev_t * mddev,
 	}
 	clear_bit(In_sync, &rdev->flags);
 	rdev->desc_nr = -1;
+	rdev->saved_raid_disk = -1;
 	err = bind_rdev_to_array(rdev, mddev);
 	if (err)
 		goto abort_export;
